src/ingameWindows/iwMilitaryBuilding.cpp

```diff
@@ -106,14 +106,25 @@ void iwMilitaryBuilding::Msg_PaintAfter()
                 (*it)->GetGOT() == GOT_NOF_DEFENDER ||
                 (*it)->GetGOT() == GOT_NOF_PASSIVESOLDIER)
         {
-            soldiers.insert(static_cast<nofSoldier*>(*it)->GetRank());
+            soldiers.insert(static_cast<nofSoldier*>(*it));
         }
     }
 
     // Soldaten zeichnen
     unsigned short i = 0;
-    for(std::multiset<unsigned>::iterator it = soldiers.begin(); it != soldiers.end(); ++it, ++i)
-        LOADER.GetMapImageN(2321 + *it)->Draw(GetX() + width_ / 2 - 22 * TROOPS_COUNT[building->nation][building->size] / 2 + 12 + i * 22, GetY() + 110, 0, 0, 0, 0, 0, 0);
+    for(std::multiset<nofSoldier*>::iterator it = soldiers.begin(); it != soldiers.end(); ++it, ++i)
+        LOADER.GetMapImageN(2321 + (*it)->GetRank())->Draw(GetX() + width_ / 2 - 22 * TROOPS_COUNT[building->nation][building->size] / 2 + 12 + i * 22, GetY() + 110, 0, 0, 0, 0, 0, 0);
+
+    // Draw health under soldiers
+    if (GAMECLIENT.GetGGS().isEnabled(ADDON_MILITARY_HITPOINTS)) { 
+        i = 0;
+        for (std::multiset<nofSoldier*>::iterator it = soldiers.begin(); it != soldiers.end(); ++it, ++i) {
```

Comment:
shouldnt this be a const_iterator? since you dont/shouldnt change the soldier itself